src/client.rs

```diff
@@ -924,91 +1119,16 @@ impl Client {
     where
         F: FnOnce(
             http::request::Builder,
-        ) -> hyper::Request<BoxBody<hyper::body::Bytes, Infallible>>,
+        ) -> hyper::Request<BoxBody<hyper::body::Bytes, Infallible>> + Send + 'static,
```

Comment:
I don't think these extra bounds are necessary?

Reply:
when running tests without `Send` bound, I get the following error -

https://github.com/jonhoo/fantoccini/blob/bce77b0d1a871ce6dafc585e3045e244a79b6218/tests/common.rs#L144  

![cargotest](https://github.com/user-attachments/assets/37f02d15-522a-4adf-9134-696af9a97e2a)

Reply:
Ah, that's because of
https://github.com/jonhoo/fantoccini/blob/5423b752d9e3b42720153b52621990e190544227/src/client.rs#L188

I think what we'll want to do here is to make the `RawRequestBuilder` generic over `<F>` and store a `Option<F>` instead of `Option<Box<dyn ...>>`. In fact, you could probably get rid of the `Option` as well, and make it:

```rust
struct RawRequestBuilder<'a, F = fn(http::request::Builder) -> hyper::Request<BoxBody<hyper::body::Bytes, Infallible>>> {
    // ...
    request_modifier: F
}

fn empty_body(req: http::request::Builder) -> hyper::Request<BoxBody<hyper::body::Bytes, Infallible>> {
    req.body(BoxBody::new(http_body_util::Empty::new()))
        .unwrap()
}

impl<'a> RawRequestBuilder<'a> {
    /// Create a new raw request builder.
    pub fn new(client: &'a Client) -> Self {
        RawRequestBuilder {
            client,
            // ...
            request_modifier: empty_body,
        }
    }
}

impl<'a, F1> RawRequestBuilder<'a, F1> {
   pub fn map_request<F2>(self, f: F2) -> RawRequestBuilder<'a, F2>
    where
        F: FnOnce(http::request::Builder) -> hyper::Request<BoxBody<hyper::body::Bytes, Infallible>>,
    {
        RawRequestBuilder {
            client: self.client,
            // ...
            request_modifier: f
        }
    }
}
```

Then in `send` you would just unconditionally call `self.map_request`.

Reply:
By the way, when including output text and such, prefer using [fenced code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks) to screenshots :)

Reply:
Thanks, The refactor makes the code more cleaner and efficient.